src/runtime_src/core/pcie/linux/shim.h

```diff
@@ -157,6 +158,7 @@ class shim
   register_xclbin(const xrt::xclbin&);
 
 private:
+  bool legacyCtxSupport;
```

Comment:
From application user can either use new h_w context or legacy LoadXclbin.
User application should not mix them. 
If application call register then it should use hw_context, which is responsible for load xclbin.
if application call load xclbin then it should use legacy opencontext/closecontext

Reply:
Yes, you're right that user app should not mix them, but what if they do? Instead, we should add the check in the driver so that we can make sure driver won't crash even when user app does something wrong.